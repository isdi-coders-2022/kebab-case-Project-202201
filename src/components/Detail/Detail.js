import styled from "styled-components";
import PropTypes from "prop-types";

const ArticleCard = styled.article`
  margin: 20px;
  left: 15px;
  top: 15px;
  background: #efeff1;
  display: flex;
  flex-direction: column;
`;

const RoundedImage = styled.img`
  position: relative;
  width: 100px;
  height: 100px;
  border-radius: 50px;
  left: 10px;
  top: 10px;
`;

const BroadcasterType = styled.p`
  width: 70px;
  height: 18px;
  padding-left: 10px;

  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;

  text-align: left;
  letter-spacing: 0.4px;
  background-color: #ffffff;
  color: #5d16c6;
`;

const NameStreamer = styled.p`
  padding-top: 20px;
  padding-left: 10px;
  text-align: left;
  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 24px;
  line-height: 16px;

  letter-spacing: 0.4px;

  color: #5d16c6;
`;

const TwitchChannel = styled.p`
  padding-left: 10px;
  text-align: right;
  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;

  text-align: left;
  letter-spacing: 0.4px;

  color: #0fa60c;
`;

const StreamerDescription = styled.p`
  padding-top: 10px;
  padding-right: 10px;
  text-align: left;

  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;
  letter-spacing: 0.4px;

  color: #000000;
`;

const ViewCount = styled.p`
  padding-top: 10px;
  padding-right: 10px;
  text-align: left;

  font-family: Roboto;
  font-style: normal;
  font-weight: 800;
  font-size: 14px;
  line-height: 16px;
  letter-spacing: 0.4px;

  color: #000000;
`;

const Detail = ({ streamerInfo, actionOnClick }) => {
  return (
    <>
      <ArticleCard className="streamerDetail" onClick={actionOnClick}>
        <RoundedImage
          className="streamerPicture"
          alt="Streamer"
          src={streamerInfo.profile_image_url}
        ></RoundedImage>
        <NameStreamer className="displayName">
          Name: {streamerInfo.display_name}
        </NameStreamer>
        <TwitchChannel className="twitchChannel">
          Twitch Channel: {streamerInfo.twitchChannel}
        </TwitchChannel>
        <StreamerDescription className="StreamerDescription">
          Description:{" "}
          {streamerInfo.description.length > 100
            ? streamerInfo.description.slice(0, 99) + "..."
            : streamerInfo.description}
        </StreamerDescription>
        <ViewCount>View Count: </ViewCount>
        <BroadcasterType className="broadcasterType">
          Contract: {streamerInfo.broadcaster_type}
        </BroadcasterType>
        <NameStreamer className="displayName">
          On Twitch for: {streamerInfo.display_name}
        </NameStreamer>
      </ArticleCard>
    </>
  );
};

Detail.propTypes = {
  display_name: PropTypes.string,
  broadcaster_type: PropTypes.string,
  description: PropTypes.string,
  profile_image_url: PropTypes.string,
  actionClick: PropTypes.func,
};

export default Detail;
