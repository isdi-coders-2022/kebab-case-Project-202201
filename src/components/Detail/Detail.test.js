import Detail from "./Detail";
import { render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";

describe("Given the Detail component", () => {
  const streamerData = {
    profile_image_url:
      "https://static-cdn.jtvnw.net/jtv_user_pictures/41b23352-5c99-4eb2-974b-1e9b8120425e-profile_image-300x300.png",
    broadcaster_type: "partner",
    display_name: "LEC",
    twitchChannel: "Channel 1",
    description:
      "Official Twitch Channel of the League of Legends European Championship. For more information, schedules and stats head to Lolesports.com",
  };

  describe("When it's rendered", () => {
    test("Then it should show 'partner' in the document", () => {
      render(<Detail streamerInfo={streamerData} />);

      const streamerElement = screen.getByText(/partner/);

      expect(streamerElement).toBeInTheDocument();
    });

    test("Then it should show 'LEC' in the document", () => {
      render(<Detail streamerInfo={streamerData}></Detail>);

      const streamerElement = screen.getByText(/Channel 1/);
      expect(streamerElement).toBeInTheDocument();
    });
  });

  describe("When it's rendered with a description shorter than 100", () => {
    test("Then it should show 'texto breve' in the document", () => {
      streamerData.description = "texto breve";

      render(<Detail streamerInfo={streamerData}></Detail>);

      const streamerElement = screen.getByText(/texto breve/);
      expect(streamerElement).toBeInTheDocument();
    });
  });

  describe("When the Streamer card is clicked", () => {
    test("Then the passed action should be called", () => {
      const mockFunction = jest.fn();

      render(
        <Detail streamerInfo={streamerData} actionOnClick={mockFunction} />
      );

      const articleElement = screen.queryByRole("article");
      userEvent.click(articleElement);

      expect(mockFunction).toHaveBeenCalled();
    });
  });

  describe("When it get a const streamer that creates a streamer with StreamerData as props", () => {
    test("then it should create and compare a snapshot with Streamer with StreamerData as component", () => {
      const streamer = TestRenderer.create(
        <Detail streamerInfo={streamerData} />
      );
      expect(streamer.toJSON()).toMatchSnapshot();
    });
  });
});
