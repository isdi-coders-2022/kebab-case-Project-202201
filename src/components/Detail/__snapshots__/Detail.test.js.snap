// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Detail component When it get a const streamer that creates a streamer with StreamerData as props then it should create and compare a snapshot with Streamer with StreamerData as component 1`] = `
<article
  className="sc-bdvvtL hhRZGJ streamerDetail"
>
  <img
    alt="Streamer"
    className="sc-gsDKAQ dZDBla streamerPicture"
    src="https://static-cdn.jtvnw.net/jtv_user_pictures/41b23352-5c99-4eb2-974b-1e9b8120425e-profile_image-300x300.png"
  />
  <p
    className="sc-hKwDye feWYtE displayName"
  >
    Name: 
    LEC
  </p>
  <p
    className="sc-eCImPb fybBUm twitchChannel"
  >
    Twitch Channel: 
    Channel 1
  </p>
  <p
    className="sc-jRQBWg lfjxiL StreamerDescription"
  >
    Description:
     
    texto breve
  </p>
  <p
    className="sc-gKclnd jeNoGN"
  >
    View Count: 
  </p>
  <p
    className="sc-dkPtRN fpdgPH broadcasterType"
  >
    Contract: 
    partner
  </p>
  <p
    className="sc-hKwDye feWYtE displayName"
  >
    On Twitch for: 
    LEC
  </p>
</article>
`;
